Enabled VSS4 support for Seat Service, fixed integration tests and DevContainer

int0x27 opened the pull request:
# VSS 4.0 support in Seat Service
- Using env variable `VSS=3` forces VSS 3.0 compliant seat position. default: `VSS=4`
- Implemented getting metadata from `sdv::databroker::v1::Broker`, `CollectorClient` renamed to `KuksaClient`
- Added support for int arrays in `createDatapoint()`
- `broker_feeder` example rewritten to actually work
- Debug dumps enhanced (e.g., `DBF_DEBUG=5 SEAT_DEBUG=5` will dump deserialized GRPC messages and statuses)
- Fixed several bugs caught from new debugs e.g., `Collector.UpdateDatapoints()` returning OK with errors.
- Improved DBF disconnection detection (previously didn't reconnect unless a feed is pending)
- Added some shared functions in `sdv::utils` namespace
- Added `DataBrokerFeeder.Ready()` function to check if we can actually feed data to databroker (connected, datapoints registered)
- DBF checks if required metadata is available and if some datapoints are missing, tries to register all of them
- Tried to address starvation in DBF process thread by busy feeding (broker feeder example)
- Although `updateDatapoint` errors are Logged (e.g. wrong type), for the moment we ignore the error, because DBF will flood databroker with errors trying to resend all.

# General changes
- Added script for updating upstream KUKSA proto files: `./tools/update-kuksa-proto.sh`
- Reduced devcontainer image layer size to allow push in ttl.sh
- `hvac_service` proto files regenerated from upstream proto files

# Integration test changes for VSS 4.0
- `integration_test/update-protobuf.sh` fixed and proto files updated from upstream
- `test_val_seat.py` updated to detect VSS3/4 mode by env. variable `USE_VSS3`
- Updated `databroker` container version to 0.4.1
- Updated `feedercan` version to 0.4.1
- Updated vscode scripts for feedercan, integration test, KUKSA client
- Fixed vscode tasks to work with recent databroker releases (`tar.gz` packed in a `.zip`)
- Removed `KDB_OPT` forcing VSS3.1 in integration
- Removed `gen_proto` dummy package in integration tests (`pytest` needs custom path to work now)
- Added Actuator target value support in `VDBHelper`
- Added test for Actuator target set usecase in `test_val_seat.py`
- Fixed `test_feeder.py` to use VSS4.0 compliant datapoints
- Updated Tesla DBC from upstream, added `DogMode` signal
- Updated `dbc2val` mapping (`DogMode` is optional as it will be missing in databroker by default)
- Fixed `it-config` for new `dbc2val` properties

int0x27 committed:
Enabled VSS4 support for seat service (VSS3 optional)

- Using env variable VSS=3, VSS3 compliant seat position is forced. default: VSS=4
- Implemented getting metadata from sdv::databroker::v1::Broker, CollectorClient renamed to KuksaClient
- Added support for int arrays in createDatapoint()
- broker_feeder example rewritten to actually work
- debug dumps enhanced (e.g. DBF_DEBUG=5 SEAT_DEBUG=5 will dump deserialized GRPC messages and statuses)
- fixed several bugs caught from new debugs (e.g. Collector.UpdateDatapoints() returning OK with errors.
- Improved databrokerfeeder disconnection detection (previously didn't reconnect unless a feed is pending)
- Added some shared functions in sdv::utils namespace
- Added DataBrokerFeeder.Ready() function to check if we can actually feed data to databroker (connected, datapoints registered)
- dbf checks if required metadata is available and if some datapoints are missing, tries to register all of them
- tried to address starvation in DBF process thread by busy feeding (broker feeder example)
- Although updateDatapoint errors are Logged (e.g. wrong type), for the moment we ignore the error, because DBF will flood databroker with errors trying to resend all.


Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

int0x27 committed:
Integration tests updated for VSS4.0

- update-protobuf.sh fixed, also added "--force" argument to get proto files from kuksa.val master
- test_val_seat.py updated to detect VSS3/4 mode by env. variable "USE_VSS3"
- bumped databroker container version to 0.4.1
- fixed vscode tasks to work with recent databroker releases (tar.gz packed in a .zip)
- removed KDB_OPT forcing VSS3.1 in integration

int0x27 committed:
Main README updated with VSS 4.0 status

- some megalinter suggestions applied
- fixed pip3 install usage

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

erikbosch reviewed (approved):
Have not run tests, but did a quick review of the changes and saw nothing problematic

int0x27 committed:
Integration test refactoring

- Added script for updating upstream kuksa proto files: `/tools/update-kuksa-proto.sh`
- Removed "gen_proto" dummy package in integration tests (pytest needs custom path to work now)
- Added Actuator target value support in `VDBHelper`
- Added test for Actuator target set usecase in `test_val_seat.py`
- Fixed `test_feeder.py` to use VSS4.0 compliant datapoints
- Updated Tesla DBC from upstream, added DogMode signal
- Updated `dbc2val` mapping (DogMode is optional as it will be missing in databroker by default)
- Updated `feedercan` version to 0.4.1
- Fixed `it-config` for new `dbc2val` properties
- Updated vscode scripts for feedercan, integration test, kukca client
- Reduced devcontainer image layer size to allow push in ttl.sh

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

SebastianSchildt commented on .github/workflows/seat_service_seatctrl_test.yml:
```diff
@@ -50,7 +50,7 @@ jobs:
       - name: cobertura-report
         if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
-        uses: 5monkeys/cobertura-action@v13
+        uses: 5monkeys/cobertura-action@18d911b62098eafd8b9db93d6c74194556cc3515
```
is there a reason we are using commit hash instead of a version here @int0x27 ? (If so, can we put a comment above why)

int0x27 replied:
They didn't have a released version, but a fix for node 12 deprecation warning here: https://github.com/5monkeys/cobertura-action/commit/18d911b62098eafd8b9db93d6c74194556cc3515
Also for security purposes it is better to pin the commit instead of using master

erikbosch replied:
It seems that the action is not that well maintained, which could give us problem in the long run. I agree that a comment would be beneficial.

int0x27 committed:
Added python3.9 in codespaces

- Improved python3.9+ detection in run-feedercan.sh
- Added comment about pinned cobertura-action revision in github actions

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

int0x27 committed:
fixed interactive apt-get error

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

int0x27 committed:
Move most of the installs in ci Dockerfile

to speedup codespaces startup

- added missing 'file' command for seat service scripts

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

int0x27 committed:
Fixed issues discovered in codespace

- python binary was missing and failed hvac testcase
- added required 'file' package that was missing in devcontainer
- removed `dbcfeeder=debug` from `it-config`, as it was too spammy
- added `feedercan.imageVersion` to `prerequisite_settings.json` - used for container version, while `feedercan.version` is used for standalone (dapr) testing
- added most of the packages installed from devcontainer Dockerfile to tools/Devcontainer to optimize codespace startup

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

int0x27 committed:
fixed arm64 devcontainer build

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

int0x27 commented:
You can test local / container integration tests in codespaces: https://github.com/codespaces/new/eclipse/kuksa.val.services/pull/103?resume=1
- wait for codespace initialization
- press `F1`
- type `Tasks: Run Task`
- Select `integration-test (containers)` or `integration-test (local)`

int0x27 committed:
Updated Seat Service README

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

int0x27 committed:
Improved seat service build scripts and readme

- added `--pack` and `--help` arguments to seat_service/build_release.sh and seat_service/build_debug.sh
- fixed readmes
- fixed build-debug.sh to use proper toolchain

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

int0x27 committed:
Fixed some issues, found by megalinter

- applied formatting patches

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

int0x27 committed:
adjusted build-release archive path to match seat service Dockerfile

Signed-off-by: Ivan Kirchev <Ivan.Kirchev@bosch.com>

SebastianSchildt reviewed (approved):
An all around renovation of some old tests and workflows. 
Also seat server compatibility with VSS4. lgtm
